code/modules/lighting/lighting_source.dm

```diff
@@ -278,7 +278,8 @@
 	L = affecting_turfs - turfs // Now-gone turfs, remove us from the affecting lights.
 	affecting_turfs -= L
 	for (var/turf/T in L)
-		T.affecting_lights -= src
+		if(T.affecting_lights && T.affecting_lights.Find(src))
```

Comment:
list.Remove(A) works 100% fine if list doesn't contain A in the first place, so the second check is not needed here atm